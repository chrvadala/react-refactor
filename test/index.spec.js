const path = require("path");
const fs = require("fs")
const {removeSpaces} = require("./testUtils");
const {patchString} = require("../lib/stringUtils");
const {execRefactor} = require("../lib/index");
const {parseFile} = require('../lib/parser')
const fixture = filename => path.join(__dirname, '__fixtures__', filename)

let example, refactoredExample;

beforeAll(done => {
  fs.readFile(fixture('example.jsx.txt'), 'utf8', (err, data) => {
    if (err) throw new Error()
    example = data
    done()
  })
})

beforeAll(done => {
  fs.readFile(fixture('refactoredExample.jsx.txt'), 'utf8', (err, data) => {
    if (err) throw new Error()
    refactoredExample = data
    done()
  })
})


describe('refactor', () => {
  it.only('should convert', () => {
    let {patch, skipped, output} = execRefactor(example)

    expect(Array.isArray(patch)).toBe(true)
    expect(Array.isArray(skipped)).toBe(true)
    expect(patch).toMatchSnapshot()
    expect(skipped).toMatchSnapshot()

    expect(removeSpaces(output))
      .toBe(removeSpaces(refactoredExample))
  })
})
