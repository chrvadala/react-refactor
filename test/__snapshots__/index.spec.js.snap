// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refactor should convert 1`] = `
Array [
  Object {
    "operation": "insert",
    "payload": "class funcComp extends React.Component {",
    "start": 0,
  },
  Object {
    "operation": "insert",
    "payload": "render()",
    "start": 0,
  },
  Object {
    "end": 25,
    "operation": "remove",
    "start": 0,
  },
  Object {
    "operation": "insert",
    "payload": "
let props = this.props;",
    "start": 26,
  },
  Object {
    "operation": "insert",
    "payload": "}",
    "start": 53,
  },
  Object {
    "operation": "insert",
    "payload": "class funcCompExport extends React.Component {",
    "start": 70,
  },
  Object {
    "operation": "insert",
    "payload": "render()",
    "start": 70,
  },
  Object {
    "end": 101,
    "operation": "remove",
    "start": 70,
  },
  Object {
    "operation": "insert",
    "payload": "
let props = this.props;",
    "start": 102,
  },
  Object {
    "operation": "insert",
    "payload": "}",
    "start": 129,
  },
  Object {
    "operation": "insert",
    "payload": "function ClassComp(props)",
    "start": 131,
  },
  Object {
    "end": 273,
    "operation": "remove",
    "start": 131,
  },
  Object {
    "operation": "insert",
    "payload": "
	let self = {props};
",
    "start": 274,
  },
  Object {
    "operation": "insert",
    "payload": "self",
    "start": 291,
  },
  Object {
    "end": 295,
    "operation": "remove",
    "start": 291,
  },
  Object {
    "operation": "insert",
    "payload": "self",
    "start": 327,
  },
  Object {
    "end": 331,
    "operation": "remove",
    "start": 327,
  },
  Object {
    "operation": "insert",
    "payload": "self",
    "start": 357,
  },
  Object {
    "end": 361,
    "operation": "remove",
    "start": 357,
  },
  Object {
    "end": 391,
    "operation": "remove",
    "start": 389,
  },
  Object {
    "operation": "insert",
    "payload": "function ClassCompExport(props)",
    "start": 400,
  },
  Object {
    "end": 548,
    "operation": "remove",
    "start": 400,
  },
  Object {
    "operation": "insert",
    "payload": "
	let self = {props};
",
    "start": 549,
  },
  Object {
    "operation": "insert",
    "payload": "self",
    "start": 566,
  },
  Object {
    "end": 570,
    "operation": "remove",
    "start": 566,
  },
  Object {
    "operation": "insert",
    "payload": "self",
    "start": 602,
  },
  Object {
    "end": 606,
    "operation": "remove",
    "start": 602,
  },
  Object {
    "operation": "insert",
    "payload": "self",
    "start": 632,
  },
  Object {
    "end": 636,
    "operation": "remove",
    "start": 632,
  },
  Object {
    "end": 666,
    "operation": "remove",
    "start": 664,
  },
]
`;

exports[`refactor should convert 2`] = `
Array [
  Object {
    "end": 708,
    "message": "This isn't a React component",
    "stack": [Error: This isn't a React component],
    "start": 668,
    "type": "FunctionDeclaration",
  },
  Object {
    "end": 760,
    "message": "This isn't a React component",
    "stack": [Error: This isn't a React component],
    "start": 710,
    "type": "FunctionDeclaration",
  },
]
`;
